src/dls_bluesky_core/core/scanspecs.py

```diff
@@ -0,0 +1,32 @@
+from typing import List, Optional
+
+import numpy as np
+from scanspec.core import Frames
+from scanspec.specs import DURATION
+
+
+def get_duration(frames: List[Frames]) -> Optional[float]:
+    """
+    Returns the duration of a number of ScanSpec frames, if known and consistent.
+
+    Args:
+        frames (List[Frames]): A number of Frame objects
+
+    Raises:
+        ValueError: If any frames do not have a Duration defined.
+
+    Returns:
+        duration (float): if all frames have a consistent duration
+        None: otherwise
+
+    """
+    for fs in frames:
+        if DURATION in fs.axes():
+            durations = fs.midpoints[DURATION]
+            first_duration = durations[0]
+            if np.all(durations == first_duration):
+                # Constant duration, return it
+                return first_duration
```

Comment:
To me the function name and docstring imply that this returns the sum of all durations, rather than the duration of a single frame. Is it meant to be one frame?

Reply:
It's meant to be used by code that expects a scanspec to have a duration for each point. The code will return one of 2 results:
- the duration if all points have the same duration
- None if duration will be present but different for each point

Reply:
Hmm, I'm not a fan from a readability point of view. 

Say the function was called `duration_per_point`, that's a better name but `None` is not an intuitive answer if there are multiple durations per point. It makes the function hard to read. Could give it an overly long name like `duration_per_point_if_all_the_same` or could make it provide a more semantically sensible answer and you can change your downstream code to deal with that. What do you think?

Reply:
How about we call it `get_constant_duration` and keep the current behaviour?

Reply:
@coretl Can I double check behaviour?
We've got frames, a list of Frames
Each Frame has midpoints, a list of AxesPoints

Right now, we iterate through the frames (except we always return during checking the first one), then return T if all the AxesPoints in that Frame are consistent, else we return None.

Should we actually be iterating through all the frames and iterating through all the AxesPoints and returning T if all the AxesPoints in ALL the Frames are consistent?

Reply:
Resolved after chat: Going to add check to make sure one and only one Frames has the Duration axis, as axes cannot be in multiple Dimensions.